### ISSUE TITLE
DeepLab and PSPNet produces nans during inference time

### ISSUE THREAD

**Issue by cygerts** [dmlc/gluon-cv#858]
Hi,

I have succesfully managed to train FCN for semantic segmentation on my custom dataset. When I change the class that I am using to any other segmentation network (so PSP, DeepLabv3 or DeepLabV3Plus) the training logs looks fine (loss decreasing) but what I noticed is that the network always predicts "nan". 

I have done some debugging and I noticed that for all segmentation networks (except from FCN) if I save the model just after the initialization the models looks fine (networks predict real numbers), but just after 1 checkpoint all mentioned networks predicts "NAN". This is even when I make the learning rate very very small. 
I am using moduleAPI for training and inference. The neural network symbol is created this way:

net = gluoncv.model_zoo.DeepLabV3Plus(...)
net.hybridize()

During training model parameters are saved using checkpoint with
arg_params, aux_params = model.get_params()

During inference:
self._model.bind(for_training=False, force_rebind=True, data_shapes=data_shapes)
self._model.load_params(weights_path)
self._model.forward(Batch(data), is_train=True)

All those steps works perfectly for FCN but fails with PSPnet and both DeepLab versions.
Using gluoncv 0.5.0

**Comment by cygerts** [comment]
Interestingly if for inference I set forward pass flag is_train=True then it works! 
Of course the results are not perfect because of the Dropout etc. but looks like some very strange behaviour is happening.
Any help would be appreciated

**Comment by cygerts** [comment]
Ok, this was the combinations of two things:
1. Bug in mxnet when testing on CPU vs training on GPU https://github.com/apache/incubator-mxnet/issues/14357
2 Training with batch_size=1
Closing the issue

**Comment by roy6324** [comment]
@cygerts 
Hello , I know this issue was closed a long time ago.

I just wanted to ask you what was the problem with training using batch_size = 1 ?
